internal/mode/static/state/graph/snippets_filter_test.go

```diff
@@ -0,0 +1,255 @@
+	invalidFilterNsName := types.NamespacedName{Namespace: "default", Name: "invalid"}
+
+	filter1 := &ngfAPI.SnippetsFilter{
+		Spec: ngfAPI.SnippetsFilterSpec{
+			Snippets: []ngfAPI.Snippet{
+				{
+					Context: ngfAPI.NginxContextMain,
+					Value:   "main snippet",
+				},
+				{
+					Context: ngfAPI.NginxContextHTTP,
+					Value:   "http snippet",
+				},
+			},
+		},
+	}
+
+	invalidFilter := &ngfAPI.SnippetsFilter{
+		Spec: ngfAPI.SnippetsFilterSpec{
+			Snippets: []ngfAPI.Snippet{
+				{
+					Context: ngfAPI.NginxContextMain,
+					Value:   "main snippet",
+				},
+				{
+					Context: "invalid context",
+					Value:   "invalid snippet",
+				},
+			},
+		},
+	}
+
+	filter2 := &ngfAPI.SnippetsFilter{
+		Spec: ngfAPI.SnippetsFilterSpec{
+			Snippets: []ngfAPI.Snippet{
+				{
+					Context: ngfAPI.NginxContextHTTPServerLocation,
+					Value:   "location snippet",
+				},
+			},
+		},
+	}
+
+	tests := []struct {
+		snippetsFilters      map[types.NamespacedName]*ngfAPI.SnippetsFilter
+		expProcessedSnippets map[types.NamespacedName]*SnippetsFilter
+		msg                  string
+	}{
+		{
+			msg:                  "no snippets filters",
+			snippetsFilters:      nil,
+			expProcessedSnippets: nil,
+		},
+		{
+			msg: "mix valid and invalid snippets filters",
+			snippetsFilters: map[types.NamespacedName]*ngfAPI.SnippetsFilter{
+				filter1NsName:       filter1,
+				invalidFilterNsName: invalidFilter,
+				filter2NsName:       filter2,
+			},
+			expProcessedSnippets: map[types.NamespacedName]*SnippetsFilter{
+				filter1NsName: {
+					Source:     filter1,
+					Conditions: nil,
+					Valid:      true,
+				},
+				filter2NsName: {
+					Source:     filter2,
+					Conditions: nil,
+					Valid:      true,
+				},
+				invalidFilterNsName: {
+					Source: invalidFilter,
+					Conditions: []conditions.Condition{staticConds.NewSnippetsFilterInvalid(
+						"spec.snippets[1].context: Unsupported value: \"invalid context\": " +
+							"supported values: \"main\", \"http\", \"http.server\", \"http.server.location\"",
+					)},
+					Valid: false,
+				},
+			},
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.msg, func(t *testing.T) {
+			g := NewWithT(t)
+
+			processedSnippetsFilters := processSnippetsFilters(test.snippetsFilters)
+			g.Expect(processedSnippetsFilters).To(BeEquivalentTo(test.expProcessedSnippets))
+		})
+	}
+}
+
+func TestValidateSnippetsFilter(t *testing.T) {
+	tests := []struct {
+		msg     string
+		filter  *ngfAPI.SnippetsFilter
+		expCond conditions.Condition
+	}{
+		{
+			msg: "valid filter",
+			filter: &ngfAPI.SnippetsFilter{
+				Spec: ngfAPI.SnippetsFilterSpec{
+					Snippets: []ngfAPI.Snippet{
+						{
+							Context: ngfAPI.NginxContextMain,
+							Value:   "main snippet",
+						},
+						{
+							Context: ngfAPI.NginxContextHTTP,
+							Value:   "http snippet",
+						},
+					},
+				},
+			},
+			expCond: conditions.Condition{},
+		},
+		{
+			msg: "invalid filter; invalid snippet context",
```

Comment:
Can we test for an invalid filter with an empty context?

Reply:
Another good question, and again, it makes me question the CRD validation. Maybe context should be required? We can't do anything with an empty context.

Reply:
Added CRD validation, go validation, and tests